analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/CompilationStartAnalysisContextExtensions.cs

```diff
@@ -0,0 +1,110 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq.Expressions;
+
+using static System.Linq.Expressions.Expression;
+using CS = Microsoft.CodeAnalysis.CSharp;
+
+namespace SonarAnalyzer.ShimLayer.AnalysisContext;
+
+public static class CompilationStartAnalysisContextExtensions
+{
+    private static readonly Action<CompilationStartAnalysisContext, Action<SymbolStartAnalysisContext>, SymbolKind> RegisterSymbolStartActionWrapper = CreateRegisterSymbolStartAnalysisWrapper();
+
+    public static void RegisterSymbolStartAction(this CompilationStartAnalysisContext context, Action<SymbolStartAnalysisContext> action, SymbolKind symbolKind) =>
+        RegisterSymbolStartActionWrapper(context, action, symbolKind);
+
+    private static Action<CompilationStartAnalysisContext, Action<SymbolStartAnalysisContext>, SymbolKind> CreateRegisterSymbolStartAnalysisWrapper()
+    {
+#pragma warning disable S103 // Lines should not be too long
+        if (typeof(CompilationStartAnalysisContext).GetMethod(nameof(RegisterSymbolStartAction)) is not { } registerMethod)
+        {
+            return static (_, _, _) => { };
```

Comment:
The same is true for SE. There we fall back to the old engine. I would say that we are good without it, but let's discuss it with @costin-zaharia-sonarsource and how he thinks about it.

Reply:
OK, I keep this discussion open to later report the outcome of the discussion with @costin-zaharia-sonarsource.

Reply:
@martin-strecker-sonarsource in which version of the compiler was the method introduced?

Reply:
Roslyn 3.0:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.compilationstartanalysiscontext.registersymbolstartaction?view=roslyn-dotnet-4.7.0#applies-to

Reply:
So we know it was delivered in Visual Studio 2019. From my point of view, it's fine if these rules are using the API will be available on Visual Studio 2019+ but I think we should mention this in the rule RSPEC.

The best would be to check with @denis-troller since this will impact our reach.